### DISCUSSION TITLE
[Feature] Timeline/home view filter

### DISCUSSION THREAD

**Discussion by aechmtwash** [immich-app/immich#3477] [Feature Request]
### The feature

Add a filter interface to the timeline/home view that can be utilized for various filters such as:

- Local, Cloud, Both
- My Assets, Partner Assets
- File Type
- Date Taken
- Person
- etc 

The filter feature can be built iteratively, initially only establishing the UI panel and a couple filter options and then expand in future iterations.  This feature could be highly useful depending on how sharing integration heads in the future (ideally having all photos you have access to show up on this view).

### Platform

- [ ] Server
- [X] Web
- [X] Mobile

**Comment by yu-jingrui** [comment]
bump this once. My need is just a dedicated view for videos.
This feature is available in synology photos:
![1738844955946](https://github.com/user-attachments/assets/8e76a95a-629a-4a13-afd8-c57fbba86cc1)

**Comment by clnhlzmn** [comment]
I would love this, especially to be able to filter by cloud/local/both.

**Comment by MartijnVanAndel** [comment]
As #12610 was closed because of this item, I'd like to add the following specific feature requests: 

- Filter based on 'tags'
- Possibility to 'exclude' based on the filter values
- Possibility to have a standard filter 'always on' on the home view.

**Comment by Novapixel1010** [comment]
## Supporting this request for separating archive content from the main timeline.

I strongly support this feature request. My specific use case is the need to keep certain types of media (e.g., thousands of short videos downloaded from TikTok, memes, or screenshots) which I need to keep, but which completely clutter the main timeline.

### The tag-based approach is ideal because:

- I can easily bulk-tag the imported archive content (e.g., tag everything as tiktok-archive).

- A simple toggle in Settings to hide anything with the tiktok-archive tag from the primary timeline would immediately solve the clutter issue.

This feature would be a massive quality-of-life improvement, allowing the main timeline to be reserved for genuine personal/family photos and videos. Thank you!

**Comment by flixhsw** [comment]
I also missed a "recently added" filter. Think it fit's well here.

**Comment by Rinyuu** [comment]
This feature would be very, very welcome. Filter options like this, or more options for folders / libraries inside existing libraries not being present is one of the main reasons I just can't switch to Immich permanently.

**Comment by philliptiongson** [comment]
yes, include screenshot to this list. (iPhones actually label their screenshots, and apple watches have very specific aspect ratios)

**Comment by charles-997** [comment]
This is an interesting and useful feature.  I would also suggest the community considers including the ability to set a "default" filter for the timeline view.

I found this as I was looking for a way to hide my RAWs from the timeline.  I view RAWs as unfinished images, however I do not want to exclude them by using an exclusion pattern on my external library settings, as if I'm looking for some photos of a specific person or from a specific event, I still want Immich to be able to search for my "unfinished RAWs".  That said, it'd be nice if they did not clutter up my timeline.

I would use this feature to filter the timeline by file type. It would be great if that filter could be applied as a default. (or if once applied, it stayed as-is until changed)
